modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/parse/MBObjectParser.java

```diff
@@ -639,8 +639,8 @@ public Expression percentage(JSONObject json, String tag, Number fallback)
             throw new MBFormatException("\"" + tag + "\" percentage from boolean " + obj
                     + " not supported, expected value between 0 and 1");
         } else if (obj instanceof JSONObject) {
-            throw new UnsupportedOperationException(
-                    "\"" + tag + "\"precentage from Function not yet supported");
+            MBFunction function = new MBFunction(this, (JSONObject) obj);
+            return function.numeric();
```

Comment:
Do we need a specific percentage function (much like we had a color function) to handle "0.5", "50%", "50" and produce an error for "1.1", "110%", "110" ?